modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloader.java

```diff
@@ -54,12 +54,14 @@
 
 class GeoIpDownloader extends PersistentTasksExecutor<PersistentTaskParams> implements ClusterStateListener {
 
+    public static final boolean GEOIP_V2_FEATURE_FLAG_ENABLED = "true".equals(System.getProperty("es.geoip_v2_feature_flag_enabled"));
+
     public static final Setting<TimeValue> POLL_INTERVAL_SETTING = Setting.timeSetting("geoip.downloader.poll.interval",
         TimeValue.timeValueDays(3), TimeValue.timeValueDays(1), Property.Dynamic, Property.NodeScope);
     public static final Setting<String> ENDPOINT_SETTING = Setting.simpleString("geoip.downloader.endpoint",
         "https://paisano.elastic.dev/v1/geoip/database", Property.NodeScope);
-    public static final Setting<Boolean> ENABLED_SETTING = Setting.boolSetting("geoip.downloader.enabled", true, Property.Dynamic,
-        Property.NodeScope);
+    public static final Setting<Boolean> ENABLED_SETTING = Setting.boolSetting("geoip.downloader.enabled", GEOIP_V2_FEATURE_FLAG_ENABLED,
+        Property.Dynamic, Property.NodeScope);
```

Comment:
Besides basing the default of this setting on the feature flag, we should additionally not register this setting if the feature flag is disabled (for now at least), by adding an if check to the `IngestGeoIpPlugin#getSettings()` method. Otherwise users could still enable the downloader, while the development of geoipv2 is still ongoing.

Reply:
That's a good catch, I'll add it